rosetta/validator/requests.go

```diff
@@ -41,33 +41,47 @@ const (
 	symbolField      = "symbol"
 	transactionField = "transaction"
 	signaturesField  = "signatures"
+
+	blockchainFailTag = "blockchain"
+	networkFailTag    = "network"
+)
+
+// Error descriptions for common errors.
+const (
+	unknownBlockchain = "network identifier has unknown blockchain field"
+	unknownNetwork    = "network identifier has unknown network field"
 )
 
-func newRequestValidator() *validator.Validate {
+func (v *Validator) newRequestValidator() *validator.Validate {
 
-	v := validator.New()
+	vl := validator.New()
 
 	// Register custom validators for known types.
 	// We register a single type per validator, so we can safely perform type
 	// assertion of the provided `validator.StructLevel` to the correct type.
-	v.RegisterStructValidation(blockValidator, identifier.Block{})
-	v.RegisterStructValidation(networkValidator, identifier.Network{})
-	v.RegisterStructValidation(accountValidator, identifier.Account{})
-	v.RegisterStructValidation(transactionValidator, identifier.Transaction{})
+	// Validation library stores the validators in a map, so if multiple
+	// validators are registered for a single type, only the last one will be used.
+	vl.RegisterStructValidation(blockValidator, identifier.Block{})
+	vl.RegisterStructValidation(accountValidator, identifier.Account{})
+	vl.RegisterStructValidation(transactionValidator, identifier.Transaction{})
+	vl.RegisterStructValidation(v.networkValidator, identifier.Network{})
 
 	// Register custom top-level validators. These validate the entire request
 	// object, compared to the ones above which validate a specific type
 	// within the request. This way we can validate some standard types (strings)
 	// or complex ones (array of currencies) in a structured way.
-	v.RegisterStructValidation(balanceValidator, rosetta.BalanceRequest{})
-	v.RegisterStructValidation(parseValidator, rosetta.ParseRequest{})
-	v.RegisterStructValidation(combineValidator, rosetta.CombineRequest{})
-	v.RegisterStructValidation(submitValidator, rosetta.SubmitRequest{})
-	v.RegisterStructValidation(hashValidator, rosetta.HashRequest{})
+	vl.RegisterStructValidation(balanceValidator, rosetta.BalanceRequest{})
+	vl.RegisterStructValidation(parseValidator, rosetta.ParseRequest{})
+	vl.RegisterStructValidation(combineValidator, rosetta.CombineRequest{})
+	vl.RegisterStructValidation(submitValidator, rosetta.SubmitRequest{})
+	vl.RegisterStructValidation(hashValidator, rosetta.HashRequest{})
```

Comment:
In order to keep it consistent, I would make it a simple function, that takes the dependency as a parameter. Then you can do `hashValidator(config)` or something.

Reply:
I changed this so that the network validation function returns a closure, so the `config` is passed as an argument. So no methods, and no need to keep the `config` as part of the validator.

I hope I didn't misunderstand what you meant 😄